workspaces/arborist/lib/add-rm-pkg-deps.js

```diff
@@ -103,9 +97,8 @@ const inferSaveType = (pkg, name) => {
   return 'prod'
 }
 
-const { hasOwnProperty } = Object.prototype
 const hasSubKey = (pkg, depType, name) => {
-  return pkg[depType] && hasOwnProperty.call(pkg[depType], name)
+  return pkg[depType] && Object.prototype.hasOwnProperty.call(pkg[depType], name)
```

Comment:
this was actually an important thing - to cache hasOwnProperty at module level. Whenever arborist is used as a library, you can't assume Object.prototype is unmodified.

Reply:
This is the same logic that made people use `void 0` instead of `undefined` this is not the kind of stuff we should have to worry about.  If you have things in your tree rewriting `Object.prototype` that's ... a different problem.

Reply:
`lib/arborist/reify.js` already calls `Object.prototype` so it's not even consistent here.  If this was something we cared about it'd be a linting rule.

Reply:
It's the same logic, and *both* are things that libraries absolutely have to worry about. If your code breaks because something did something dumb elsewhere, it's still broken.

This isn't something that's trivial to lint against.